src/dungeon/user.rs

```diff
@@ -0,0 +1,184 @@
+use super::Path;
+use std::collections::HashSet;
+
+extern crate serde_derive;
+use serde_derive::Deserialize;
+
+/// Tracks what dungeons a user has ran since the last daily reset
+/// and from the last Dungeon Frequenter achievement completion.
+#[derive(Debug, Clone)]
+pub struct UserProgress {
+    dungeon_frequenter_progress: HashSet<u8>,
+    dungeons_ran_today: HashSet<String>,
+}
+
+#[derive(Deserialize)]
+struct AchievementProgress {
+    id: u32,
+    bits: Option<Vec<u8>>,
+}
+
+const DUNGEON_FREQUENTER_ID: u32 = 2963;
```

Comment:
Where does this number come from? comment plz

Reply:
It is the achievement id of the Dungeon Frequenter achievement.

Reply:
I think a comment should be added to the file for this.

Reply:
comment added